src/main/java/com/faforever/moderatorclient/api/FafApiCommunicationService.java

```diff
@@ -99,7 +104,14 @@ private void authorize(String username, String password) {
         details.setUsername(username);
         details.setPassword(password);
 
-        restOperations = restTemplateBuilder.configure(new OAuth2RestTemplate(details));
+        restTemplate = restTemplateBuilder.configure(new OAuth2RestTemplate(details));
+        restTemplate.setInterceptors(Collections.singletonList(
```

Comment:
I guess this will set it globally overriding anything we set in request right? Do we really want this? I would prefer to just modify  POST and PATCH methods.

Reply:
The only place where we do not send application/json is the login (form encoded). Even there the code works. I see no reason to set it per method.

Reply:
I mean something like this
```
        restTemplate.setInterceptors(Collections.singletonList(
                (request, body, execution) -> {
                    if (request.getMethod() == HttpMethod.POST || request.getMethod() == HttpMethod.PATCH || request.getMethod() == HttpMethod.PUT) {
                        HttpHeaders headers = request.getHeaders();
                        headers.setContentType(MediaType.APPLICATION_JSON);
                    }
                    return execution.execute(request, body);
                }
        ));
```

Reply:
I would agree to @bukajsytlos just because it works does not mean we should do it. So if in the future somebody changes the MeidaTypes in the api he can be sure they are all set correctly.

Reply:
The formatting is ugly, but I accept this as the better solution.